src/perl/lib/wtsi_clarity/irods/irods_publisher.pm

```diff
@@ -60,6 +66,26 @@ sub _execute_irods_command {
   return $exit_code;
 }
 
+sub _save_hash {
+  my ($self, $filename, $location) = @_;
+
+  my $command = "ichksum $filename";
+
+  my ($writer, $reader, $err);
+  open3($writer, $reader, $err, $command);
+  my $output = <$reader>;
+
+  $output =~ /$filename\s+(\w+)/smx;
+  if ($output) {
+    my $md5_hash = $1;
+    $self->insert_hash_to_database($filename, $md5_hash, $location);
```

Comment:
I think this class is doing more than one thing.
It is responsible to publish a file to iRODS and also to add a record to a DB with the published file's checksum.
I would have a public attribute with the value of the hash and another class would write the hash into the DB. For example the report classes (manifest, sample_qc_report). This way we can separate it and we can also have a report publish to iRODS without saving its checksum to the DB.